### PR TITLE
Refactor Betweenness

### PR DESCRIPTION
**Pull request by angriman**
This PR brings the following changes to `Betweenness`:
- Fix the normalization for unweighted graphs as suggested in #592 
- Faster parallel performances by aggregating the vertex and edge scores using `omp atomic` operations instead of using thread-local vectors as done in #607 
- Parallel score normalization
- Additional tests for the normalized betweenness

### CODE REVIEW

**Review by fabratu — APPROVED** [review]
LGTM.

File: networkit/cpp/centrality/Betweenness.cpp [hunk 1]
```diff
@@ -30,20 +30,9 @@ void Betweenness::run() {
...
-    if (computeEdgeCentrality) {
-        edgeScorePerThread.resize(maxThreads, std::vector<double>(G.upperEdgeIdBound()));
-    }
-    DEBUG("edge score per thread: ", edgeScorePerThread.size());
-
-    std::vector<std::vector<double>> dependencies(maxThreads, std::vector<double>(z));
+    std::vector<std::vector<double>> dependencies(omp_get_max_threads(), std::vector<double>(z));
     std::vector<std::unique_ptr<SSSP>> sssps;
-    sssps.resize(maxThreads);
+    sssps.resize(omp_get_max_threads());
```
Comment (michitux) [inline networkit/cpp/centrality/Betweenness.cpp:35]: Similar to #607, the `SSSP` instance could instead be initialized here without the global `sssps` vector.

File: networkit/cpp/centrality/Betweenness.cpp [hunk 2]
```diff
@@ -59,68 +48,53 @@ void Betweenness::run() {
...
             scoreData[u] = scoreData[u] / pairs;
         });
-        if (computeEdgeCentrality) {
-            for (count edge = 0; edge < edgeScoreData.size(); edge++) {
-                edgeScoreData[edge] =  edgeScoreData[edge] / edges;
-            }
-        }
+
+        if (computeEdgeCentrality)
+            G.parallelForEdges([&](node, node, edgeweight, edgeid eid) {
```
Comment (michitux) [inline networkit/cpp/centrality/Betweenness.cpp:95]: I have not tested this, but using `G.parallelForEdges` here seems like a total waste of running time unless really a lot of edges have been deleted as this iterates over all nodes and their outgoing edges which involves a lot of random memory accesses compared to the simple sweep of the `edgeScoreData` vector. I would suggest to instead use the old loop with a simple `#pragma omp parallel for`.

**Review by michitux — CHANGES_REQUESTED** [review]

File: networkit/cpp/centrality/Betweenness.cpp (hunk 2 above)
Comment (angriman) [reply networkit/cpp/centrality/Betweenness.cpp]: Good point, thanks. Amended as suggested.

File: networkit/cpp/centrality/Betweenness.cpp [hunk 3]
```diff
@@ -59,68 +48,53 @@ void Betweenness::run() {
...
-        if (computeEdgeCentrality) {
-            for (count edge = 0; edge < edgeScoreData.size(); edge++) {
-                edgeScoreData[edge] =  edgeScoreData[edge] / edges;
-            }
-        }
+
+        if (computeEdgeCentrality)
+#pragma omp parallel for
+            for (omp_index i = 0; i < static_cast<omp_index>(edgeScoreData.size()); ++i)
+                edgeScoreData[i] =  edgeScoreData[i] / static_cast<double>(edges);
```
Comment (michitux) [inline networkit/cpp/centrality/Betweenness.cpp:97]: For the sake of readability I would suggest to add some braces around these code blocks, in particular with the `#pragma` that is not indented this looks weird.

**Review by michitux — APPROVED** [review]

File: networkit/cpp/centrality/Betweenness.cpp (hunk 3 above)
Comment (angriman) [reply networkit/cpp/centrality/Betweenness.cpp:97]: This block might look suspicious indeed, I added braces.

**Review by michitux — APPROVED** [review]

**Pull request merged**